### PR TITLE
[FLINK-8217] [Kinesis connector] Properly annotate APIs of flink-connector-kinesis

### PR DESCRIPTION
**Pull request by bowenli86**
## What is the purpose of the change

Properly annotate classes in flink-connector-kinesis

## Brief change log

Properly annotate classes in flink-connector-kinesis

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

none

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

**Comment by greghogan** [comment]
Changes to the public API require a [FLIP](https://cwiki.apache.org/confluence/display/FLINK/Flink+Improvement+Proposals).

**Comment by bowenli86** [comment]
@greghogan this is a subtask of FLINK-8192. 

@tzulitai What do you think of having a FLIP to FLINK-8192?

**Comment by tzulitai** [comment]
@bowenli86 @greghogan 
To my understanding, the proposed changes for FLINK-8192 simply suggests adding proper annotations to the APIs in the Flink connectors. This does not count as a public API change, no?

For example, classes such as `KinesisStreamShard` or `SequenceNumber` should be properly annotated as `@Internal`. I understand that by doing so we're narrowing down the "scope" of the public API, but I don't think there's was previously any way that those classes could have been used with the connector in any meaningful way anyways.

File: flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/model/KinesisStreamShard.java [hunk 1]
```diff
@@ -24,9 +24,13 @@
...
- * A legacy serializable representation of a AWS Kinesis Stream shard. It is basically a wrapper class around the information
- * provided along with {@link com.amazonaws.services.kinesis.model.Shard}.
+ * A legacy serializable representation of a AWS Kinesis Stream shard.
+ * It is basically a wrapper class around the information provided along
+ * with {@link com.amazonaws.services.kinesis.model.Shard}.
+ *
+ * @deprecated Will be remove in a future version in favor of {@link StreamShardHandle}.
  */
+@Deprecated
 public class KinesisStreamShard implements Serializable {
```
Comment (tzulitai) [inline flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/model/KinesisStreamShard.java]: Although this is deprecated, we should probably also annotate this as `@Internal`

File: flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/serialization/KinesisDeserializationSchemaWrapper.java [hunk 2]
```diff
@@ -27,6 +28,7 @@
  *
  * @param <T> The type created by the deserialization schema.
  */
+@PublicEvolving
 public class KinesisDeserializationSchemaWrapper<T> implements KinesisDeserializationSchema<T> {
```
Comment (tzulitai) [inline flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/serialization/KinesisDeserializationSchemaWrapper.java]: I think this is also only meant for `@Internal` use.

**Comment by bowenli86** [comment]
I agree with @tzulitai . FLINK-8192 acknowledges some un-annotated APIs as 'public', not necessarily changing existing official public APIs

**Comment by greghogan** [comment]
@bowenli86 the `@Public` annotation is much more than an acknowledgement, it promises that Flink will support that API essentially forever (despite the large number of "2.0" tickets, it's likely that few APIs will be modified). If there is an implicit annotation I think it is closer to `@Internal` than to `@PublicEvolving`, and either of these two can be used with care without buy-in from the full developer community.

I'm not sure that a FLIP is the best way to handle the promotion of `@PublicEvolving` to `@Public`. I think this might be better handled once per release, and there are other "bulk" considerations as well such as dependency versions which we may want to formalize in the release schedule.

**Comment by StephanEwen** [comment]
I agree that we should be careful with `@Public` and if in doubt resort to `@PublicEvolving` (and use `@Internal` sort of as the default).

Given that @tzulitai did a good amount of the Kinesis work, his buy-in for `@Public` classes would be good.

**Comment by tzulitai** [comment]
My intention for opening FLINK-8217 was more towards the purpose of clearly identifying what is `@Internal` and should not be used. I can agree with that we at least separate APIs to be annotated as either `@Internal` or `@PublicEvolving` for now. Whether or not we want to elevate certain APIs to `@Public` can probably be a separate issue.

**Comment by bowenli86** [comment]
A good point from Gordon. This PR's main purpose is actually reminding users which class to use and which not. `PublicEvolving` should be good enough.

Will update the PR

**Comment by bowenli86** [comment]
@tzulitai  The PR has been updated

**Comment by tzulitai** [comment]
I was waiting for any objections from others before moving ahead to merge this.
If there are no other objections, will merge by the end of the week :)

**Comment by tzulitai** [comment]
Since there are no other objections, will merge this to `master` now ..

**Pull request closed without merging**